### ISSUE TITLE
Bug: Training behaves very differently depending on whether a validation set is included

### ISSUE THREAD

**Issue by rafwaf** [keras-team/keras#13999]
**System information**  
- Have I written custom code (as opposed to using example directory):  Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Windows 10
- TensorFlow backend (yes / no): Yes 
- TensorFlow version:  2.1.0 (Can't upgrade yet due to compatibility issues)
- Keras version:  2.2.4-tf
- Python version:  3.7.6
- CUDA/cuDNN version:  10.1/6.14.11.10010
- GPU model and memory:  Nvidia Quadro M2000 4GB Memory

**Describe the current behavior**  
When training the model with only a training dataset, the loss and metrics steadily converge to a perfect score implying that the model was successfully able to overfit the training data (training loss of very close to zero and Jaccard_index(similar to accuracy, 0 is worst, 1 is the best) of very close to one. But to my surprise, when I added a validation set to the fit function, all of a sudden, the training loss and Jaccard_index were "stuck" and converged to a much worse score (jaccard_index didn't cross the 0.3 mark).
I'm including the loss curves (actually the jaccard_index curves) from both scenarios of training, with and without a validation set, as an image below.

**Describe the expected behavior**  
I would expect the training loss (and Jaccard_index metric) to be able to overfit on the training data and be able to reach the same near-perfect score, since the training data has not changed. For what it's worth, I'll mention that I have batchNorm, Parametric ReLU in my CNN.

**Code to reproduce the issue**  
Provide a reproducible test case that is the bare minimum necessary to generate the problem.  
```python
from numpy.random import seed
seed(101)
from tensorflow.compat.v1 import set_random_seed
set_random_seed(102)

import tensorflow as tf
import tensorflow.keras as keras
from tensorflow.keras.layers import Input, Dense, Conv2D, MaxPooling2D, Lambda, BatchNormalization #PReLU
from tensorflow.keras.models import Model, Sequential
from tensorflow.keras.initializers import Constant
import numpy as np
from tensorflow.keras.preprocessing.image import ImageDataGenerator
import os
import tensorflow.keras.backend as K
import numpy as np
from matplotlib import pyplot as plt
import cv2
from tensorflow.keras.optimizers import SGD, Adadelta

def jaccard_index(y_true, y_pred):
    intersection = K.sum(K.abs(y_true * y_pred))
    sum_ = K.sum(K.abs(y_true) + K.abs(y_pred))
    jac = (intersection) / (sum_ - intersection)
    return (jac)

def mask_crop_training_data_generator(x, y):
     while True:
        batch_x = next(x)
        batch_y = next(y)
        yield (batch_x, batch_y)

IMAGE_HEIGHT = 85
IMAGE_WIDTH = 128
NUMBER_OF_COLOR_CHANNELS = 1

########################################    With Parametric Relu and Batch Normalization (shared axes prelu and 0.25 initalizer)    ###########################
# build the full segmentation network
# block 1
my_layers_with_prelu_and_batch_normalization = [Input(shape=(IMAGE_HEIGHT, IMAGE_WIDTH, NUMBER_OF_COLOR_CHANNELS), dtype="float32", name="main_input"),
Conv2D(filters=32, kernel_size=(25,25), strides=(1,1), padding="same", name="conv_2D_1"),
BatchNormalization(name="batch_normalization_1_5", momentum=0.8),
tf.keras.layers.PReLU(shared_axes=[1, 2], alpha_initializer=Constant(value=0.25)),
Conv2D(filters=32, kernel_size=(25,25), strides=(1,1), padding="same", name="conv_2D_2"),
BatchNormalization(name="batch_normalization_2_5", momentum=0.8),
tf.keras.layers.PReLU(shared_axes=[1, 2], alpha_initializer=Constant(value=0.25)),
MaxPooling2D(name="max_pooling_2D_3"),
# block 2
Conv2D(filters=64, kernel_size=(13,13), strides=(1,1), padding="same", name="conv_2D_4"),
BatchNormalization(name="batch_normalization_4_5", momentum=0.8),
tf.keras.layers.PReLU(shared_axes=[1, 2], alpha_initializer=Constant(value=0.25)),
Conv2D(filters=64, kernel_size=(13,13), strides=(1,1), padding="same", name="conv_2D_5"),
BatchNormalization(name="batch_normalization_5_5", momentum=0.8),
tf.keras.layers.PReLU(shared_axes=[1, 2], alpha_initializer=Constant(value=0.25)) ,
Conv2D(filters=64, kernel_size=(13,13), strides=(1,1), padding="same", name="conv_2D_6"),
BatchNormalization(name="batch_normalization_6_5", momentum=0.8),
tf.keras.layers.PReLU(shared_axes=[1, 2], alpha_initializer=Constant(value=0.25)) ,
MaxPooling2D(name="max_pooling_2D_7"),
# block 3
Conv2D(filters=64, kernel_size=(7,7), strides=(1,1), padding="same", name="conv_2D_8"),
BatchNormalization(name="batch_normalization_8_5", momentum=0.8),
tf.keras.layers.PReLU(shared_axes=[1, 2], alpha_initializer=Constant(value=0.25)),
Conv2D(filters=64, kernel_size=(7,7), strides=(1,1), padding="same", name="conv_2D_9") ,
BatchNormalization(name="batch_normalization_9_5", momentum=0.8),
tf.keras.layers.PReLU(shared_axes=[1, 2], alpha_initializer=Constant(value=0.25)),
Conv2D(filters=64, kernel_size=(7,7), strides=(1,1), padding="same", name="conv_2D_10") ,
BatchNormalization(name="batch_normalization_10_5", momentum=0.8),
tf.keras.layers.PReLU(shared_axes=[1, 2], alpha_initializer=Constant(value=0.25)),
Conv2D(filters=64, kernel_size=(7,7), strides=(1,1), padding="same", name="conv_2D_11") ,
BatchNormalization(name="batch_normalization_11_5", momentum=0.8),
tf.keras.layers.PReLU(shared_axes=[1, 2], alpha_initializer=Constant(value=0.25)),
MaxPooling2D(name="max_pooling_2D_12"),
# final steps
Conv2D(filters=1024, kernel_size=(5,5), strides=(1,1), padding="same", name="conv_2D_13") ,
tf.keras.layers.PReLU(shared_axes=[1, 2], alpha_initializer=Constant(value=0.25)),
BatchNormalization(name="batch_normalization_12_5", momentum=0.8),
Conv2D(filters=1, kernel_size=(1,1), strides=(1,1), padding="same", activation="sigmoid", name="segmentation_mask")]

segmentation_model = Sequential()

for layer in my_layers_with_prelu_and_batch_normalization:
    segmentation_model.add(layer)
print(segmentation_model.summary())

train_img_datagen = ImageDataGenerator(
        rescale=1./255,
        vertical_flip=True,
        horizontal_flip=True,
        rotation_range=0)
train_mask_datagen = ImageDataGenerator(
        # rescale=1./255,
        vertical_flip=True,
        horizontal_flip=True,
        rotation_range=0,
        preprocessing_function=inverse_mask)
        
val_img_datagen = ImageDataGenerator(rescale=1./255,
        vertical_flip=True,
        horizontal_flip=True,)
val_mask_datagen = ImageDataGenerator(#rescale=1./255, 
        vertical_flip=True,
        horizontal_flip=True,
        preprocessing_function=inverse_mask)

epochs = 1000000
check_point_loss= "jaccard_index"
number_of_batches = 16
from datetime import datetime
date_and_time = datetime.now().strftime("%Y-%m-%d--%H%M")
save_name = "crop_train_" + check_point_loss + "_" + str(epochs) + "_epochs_" + str(number_of_batches) + "_batches_gpu" + date_and_time
checkpoint_path = "checkpoints\\" + save_name
log_dir = "logs\\fit\\" + save_name
try:
    os.mkdir(checkpoint_path)
    # os.mkdir(checkpoint_path + "\\logs")
    # os.mkdir(checkpoint_path + "\\logs\\fit")
except:
    print("wasn't able to create the checkpoint folder. Exiting now..............")
    exit()

checkpoint_dir = os.path.dirname(checkpoint_path)

# Create a callback that saves the model's weights
cp_callback = keras.callbacks.ModelCheckpoint(filepath=checkpoint_path,
                                                monitor=check_point_loss,
                                                save_best_only=True,
                                                save_weights_only=False,
                                                verbose=1,
                                                mode='max')

tb_callback = keras.callbacks.TensorBoard(log_dir=log_dir, histogram_freq=1,
                                            write_graph=True,
                                            write_images=True,
                                            update_freq='epoch',
                                            profile_batch=0)
                                            # write_grads=True)

early_stopping_callback = keras.callbacks.EarlyStopping(monitor=check_point_loss, 
                                                                        patience=1000, 
                                                                        mode="max",
                                                                        restore_best_weights=True)

train_image_generator = train_img_datagen.flow_from_directory(
"",
batch_size = number_of_batches, target_size=(85, 128), shuffle=True, class_mode=None, color_mode="grayscale", seed=101)

train_mask_generator = train_mask_datagen.flow_from_directory(
"",
batch_size = number_of_batches, target_size=(10, 16), color_mode="grayscale", shuffle=True, class_mode=None, seed=101, interpolation="nearest")

val_image_generator = val_img_datagen.flow_from_directory(
"",
batch_size = 16, target_size=(85, 128), shuffle=True, class_mode=None, color_mode="grayscale", seed=101)


val_mask_generator = val_mask_datagen.flow_from_directory(
"",
batch_size = 16, target_size=(10, 16), color_mode="grayscale", shuffle=True, class_mode=None, seed=101, interpolation="nearest")



train_generator = crop_generator_masks(train_image_generator, train_mask_generator, 256, 256)
val_generator = crop_generator_masks(val_image_generator, val_mask_generator, 256, 256)

bla = mask_crop_training_data_generator(train_image_generator, train_mask_generator)
val_bla = mask_crop_training_data_generator(val_image_generator, val_mask_generator)

my_sgd = SGD(momentum=0.9, nesterov=True)
segmentation_model.compile(loss="binary_crossentropy",
              optimizer=my_sgd,
              metrics=[jaccard_index])

results = segmentation_model.fit(bla,  steps_per_epoch=3, epochs=epochs, 
                          validation_data=val_bla,
                          callbacks=[cp_callback, tb_callback, early_stopping_callback], validation_steps=1)
```

**Other info / logs**  
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.  
![image](https://user-images.githubusercontent.com/9952670/80145991-f2ce0300-8565-11ea-9144-e40c6d199c1e.png)

Adding another image of a few other attempts to tweak showing the difference between good scores and bad scores, the main difference between good scores(overfit achieved) and bad scores (overfit not aciheved) being whether or not a validation set was included with the fit function:
![image](https://user-images.githubusercontent.com/9952670/80425027-ba953000-8897-11ea-98ca-4aba9a43e06f.png)

**Comment by rafwaf** [comment]
@saikumarchalla Hi, I see m request was tagged as an implementation issue, though I believe it actually reflects a bug. 

I don't see any reason why including or not including a validation set should change the training loss in any way. Neither affect the speed in which it converges nor the ultimate value to which it converges, nor even whether or not it is able to over fit. 

Please, if you can, reconsider labeling this as a bug so that it can be appropriately looked upon. Otherwise I believe a real bug, with significant consequences will slip through.

Thank you.

**Comment by jvishnuvardhan** [comment]
@rafwaf This is a stale issue. Sorry for late response. 

Is this still an issue for you? Can you please share a simple standalone code to reproduce the issue? Simplify the code as much as possible. Also, check your code with the recent TF versions (`2.6` and `tf-nightly`) and let us know whether the issue persists with recent TF versions. Thanks!
